libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -169,12 +170,18 @@ public FIXConnection(ReadableByteChannel inboundChannel, GatheringByteChannel ou
 
         this.adminMessage = new FIXMessage(ADMIN_MESSAGE_MAX_FIELD_COUNT, config.getFieldCapacity());
 
-        this.currentTimeMillis = 0;
-
         this.currentTime = new FIXTimestamp();
 
         this.currentTimestamp = new FIXValue(CURRENT_TIMESTAMP_FIELD_CAPACITY);
-
+        
+        this.lastRxMillis = initialMillis;
+        
+        this.lastTxMillis = initialMillis;
+        
+    	this.currentTimeMillis = initialMillis;
+    	
+    	this.currentTime.setEpochMilli(initialMillis);
+    	
```

Comment:
Let's minimize the diff in this method.

Reply:
minimized in [29ebc04](https://github.com/paritytrading/philadelphia/pull/302/commits/29ebc049f7e659daabc7aca7264b2a1bf0820b9d)